qa/acceptance-tests/src/test/java/io/camunda/it/client/RolesByTenantIntegrationTest.java

```diff
@@ -0,0 +1,222 @@
+    // when
+    camundaClient.newAssignRoleToTenantCommand(tenantId).roleId(roleId).send().join();
+    // then
+    Awaitility.await("Role should be visible in tenant role search")
+        .atMost(TIMEOUT_DATA_AVAILABILITY)
+        .untilAsserted(
+            () -> {
+              final var roles = camundaClient.newRolesByTenantSearchRequest(tenantId).send().join();
+              assertThat(roles.items())
+                  .anySatisfy(
+                      role -> {
+                        assertThat(role.getRoleId()).isEqualTo(roleId);
+                        assertThat(role.getName()).isEqualTo(roleName);
+                        assertThat(role.getDescription()).isEqualTo(roleDesc);
+                      });
+            });
+  }
+
+  @Test
+  void shouldRejectIfTenantDoesNotExist() {
+    // given
+    final var nonExistingTenantId = "tenant-" + Strings.newRandomValidIdentityId();
+    final var roleId = "role-" + Strings.newRandomValidIdentityId();
+    final var roleName = "name-" + Strings.newRandomValidIdentityId();
+    final var roleDesc = "desc-" + Strings.newRandomValidIdentityId();
+    createRole(roleId, roleName, roleDesc);
+    // when and then
+    assertThatThrownBy(
+            () ->
+                camundaClient
+                    .newAssignRoleToTenantCommand(nonExistingTenantId)
+                    .roleId(roleId)
+                    .send()
+                    .join())
+        .isInstanceOf(ProblemException.class)
+        .hasMessageContaining("404: 'Not Found'")
+        .hasMessageContaining("no tenant with this ID exists");
+  }
+
+  @Test
+  void shouldRejectIfRoleDoesNotExist() {
+    final var tenantId = "tenant-" + Strings.newRandomValidIdentityId();
+    final var roleId = "role-" + Strings.newRandomValidIdentityId();
+    createTenant(tenantId);
+    assertThatThrownBy(
+            () -> camundaClient.newAssignRoleToTenantCommand(tenantId).roleId(roleId).send().join())
+        .isInstanceOf(ProblemException.class)
+        .hasMessageContaining("404: 'Not Found'")
+        .hasMessageContaining("role doesn't exist");
+  }
+
+  @Test
+  void shouldRejectIfRoleAlreadyAssigned() {
+    // given
+    final var tenantId = "tenant-" + Strings.newRandomValidIdentityId();
+    final var roleId = "role-" + Strings.newRandomValidIdentityId();
+    final var roleName = "name-" + Strings.newRandomValidIdentityId();
+    final var roleDesc = "desc-" + Strings.newRandomValidIdentityId();
+    createTenant(tenantId);
+    createRole(roleId, roleName, roleDesc);
+    camundaClient.newAssignRoleToTenantCommand(tenantId).roleId(roleId).send().join();
+    Awaitility.await("Role should be assigned before retrying")
+        .atMost(TIMEOUT_DATA_AVAILABILITY)
+        .untilAsserted(
+            () -> {
+              final var roles = camundaClient.newRolesByTenantSearchRequest(tenantId).send().join();
+              assertThat(roles.items()).anyMatch(r -> r.getRoleId().equals(roleId));
+            });
```

Comment:
🔧 I think we can get rid of this await. The assignment command goes to the engine where the commands are processed sequentially. Waiting until it's available in secondary storage is unnecessary.